packages/superset-ui-connection/src/types.ts

```diff
@@ -11,11 +11,18 @@ export type FetchRetryOptions = {
 };
 export type Headers = { [k: string]: string };
 export type Host = string;
+export type JSONPrimitive = string | number | boolean | null;
+export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
+export type JSONObject = { [member: string]: JSONValue };
+export type JSONArray = JSONValue[];
+
+// `JSONObject` could not accept more specific types, so we had to employ any.
+// see: https://github.com/microsoft/TypeScript/issues/15300
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export type Json = { [k: string]: any };
+export type JsonObject = { [member: string]: any };
```

Comment:
(optional) can also do `Record<string, any>`

Reply:
```ts
export type JSONValue = JSONPrimitive | JSONObject | JSONArray;
export type JSONObject = Record<string, JSONValue>;
```

This will generate following error:

```
Type alias 'JSONObject' circularly references itself.  ts(2456)
```